lang/src/org/partiql/lang/eval/visitors/StaticTypeVisitorTransform.kt

```diff
@@ -0,0 +1,532 @@
+            }
+        }
+
+        private fun PartiqlAst.Expr.Id.toPathExpr(): PartiqlAst.PathStep.PathExpr =
+            PartiqlAst.build {
+                pathExpr(index = lit(ion.newString(name.text).toIonElement(), this@toPathExpr.extractSourceLocation()), case = case)
+            }
+
+        private fun errUnboundName(name: String, metas: MetaContainer): Nothing = throw SemanticException(
+            "No such variable named '$name'",
+            ErrorCode.SEMANTIC_UNBOUND_BINDING,
+            propertyValueMapOf(
+                Property.BINDING_NAME to name
+            ).addSourceLocation(metas)
+        )
+
+        private fun errIllegalGlobalVariableAccess(name: String, metas: MetaContainer): Nothing = throw SemanticException(
+            "Global variable access is illegal in this context",
+            ErrorCode.SEMANTIC_ILLEGAL_GLOBAL_VARIABLE_ACCESS,
+            propertyValueMapOf(
+                Property.BINDING_NAME to name
+            ).addSourceLocation(metas)
+        )
+
+        private fun errAmbiguousName(name: String, metas: MetaContainer): Nothing = throw SemanticException(
+            "A variable named '$name' was already defined in this scope",
+            ErrorCode.SEMANTIC_AMBIGUOUS_BINDING,
+            propertyValueMapOf(
+                Property.BINDING_NAME to name
+            ).addSourceLocation(metas)
+        )
+
+        private fun errUnimplementedFeature(name: String, metas: MetaContainer? = null): Nothing = throw SemanticException(
+            "Feature not implemented yet",
+            ErrorCode.UNIMPLEMENTED_FEATURE,
+            propertyValueMapOf(
+                Property.FEATURE_NAME to name
+            ).also {
+                if (metas != null) {
+                    it.addSourceLocation(metas)
+                }
+            }
+        )
+
+        private fun addLocal(name: String, type: StaticType, metas: MetaContainer) {
+            val existing = localsOnlyEnv[BindingName(name, BindingCase.INSENSITIVE)]
+            if (existing != null) {
+                errAmbiguousName(name, metas)
+            }
+            localsMap[name] = type
+            // this requires a new instance because of how [Bindings.ofMap] works
+            localsOnlyEnv = wrapBindings(Bindings.ofMap(localsMap), currentScopeDepth)
+        }
+
+        override fun transformExprCallAgg(node: PartiqlAst.Expr.CallAgg): PartiqlAst.Expr {
+            return PartiqlAst.build {
+                callAgg_(
+                    // do not transform the funcExpr--as this is a symbolic name in another namespace (AST is over generalized here)
+                    node.setq,
+                    node.funcName,
+                    transformExpr(node.arg),
+                    transformMetas(node.metas))
+            }
+        }
+
+        // TODO: find if this override shouldn't be necessary because PIG's Call doesn't store the function call name as the first arg
```

Comment:
yeah, this is basically the same as the method in the super class now right?  If so, just delete this.

Reply:
Got it. Removed the comment + override.